app/controller/grid/Grid.js

```diff
@@ -0,0 +1,328 @@
+/**
+ * This class is the controller for the cpsi mapview WFS
+ * generic grid class
+ *
+ */
+Ext.define('CpsiMapview.controller.grid.Grid', {
+    extend: 'Ext.app.ViewController',
+    alias: 'controller.cmv_grid',
+
+    requires: [
+        'Ext.menu.Menu',
+        'Ext.grid.filters.Filters',
+        'BasiGX.util.Layer',
+        'GeoExt.util.OGCFilter'
+    ],
+
+    /**
+    * The currently active spatial filter for the layer.
+    *
+    * @cfg {Ext.util.Filter} spatialFilter
+    */
+    spatialFilter: null,
+
+    /**
+     * Fire a onRowDblClick to be handled by the parent application
+     */
+    onRowDblClick: function (sender, record) {
+        this.fireEvent('onRowDblClick', sender, record);
+    },
+
+    /**
+     * Zoom the map to the selected feature with a buffer
+     *
+     * @param {ol.Feature} feature
+     * @param {ol.Map} map
+     * @private
+     */
+    zoomToFeature: function (feature, map) {
+
+        // TODO check for feature type when zooming
+        var extent = feature.getGeometry().getExtent();
+        // as this is a point then buffer it by 100m
+        extent = ol.extent.buffer(extent, 100);
+        //map.getView().fit(extent, map.getSize());
+
+        var view = map.getView();
+
+        var resolution = view.getResolutionForExtent(extent);
+        var zoom = view.getZoomForResolution(resolution);
+        var center = ol.extent.getCenter(extent);
+        var duration = 2000;
+
+        view.animate({
+            center: center,
+            duration: duration
+        });
+
+        view.animate(
+            {
+                zoom: zoom - 1,
+                duration: duration / 2
+            }, {
+                zoom: zoom,
+                duration: duration / 2
+            }
+        );
+    },
+
+    /**
+     * Open a row-level context-menu with a Zoom to Feature option
+     * @private
+     */
+    onItemContextMenu: function (grid, record, item, index, e) {
+
+        var me = this;
+        var map = me.getViewModel().get('map');
+
+        if (!me.contextMenu) {
+            me.contextMenu = Ext.create('Ext.menu.Menu', {
+                items: [{
+                    text: 'Zoom to Feature',
+                    scope: me,
+                    handler: function () {
+                        var record = grid.getSelection()[0];
```

Comment:
This should be the `record` variable provided in the `onItemContextMenu` listener function.

Reply:
The record always seems to be available when `onItemContextMenu` is called but is then undefined in the handler. I can set me.record in the `onItemContextMenu` function and access this in the handler, but this doesn't seem the cleanest approach. I'm unsure why the variable isn't available in the anonymous function.

Reply:
Updated